File: internal/client/src/message.ts
export interface Message {
	init?: MessageInit
	segment?: MessageSegment
	beat?: MessageBeat
}

export interface MessageInit {
	id: string
}

export interface MessageSegment {
	init: string // id of the init segment
	timestamp: number // presentation timestamp in milliseconds of the first sample
	// TODO track would be nice
}

export interface MessageBeat {
	timestamp: number // presentation timestamp in milliseconds
}

export interface Debug {
	max_bitrate: number
}

File: qwantify/internal/warp/message.go
package warp

type Message struct {
	Init    *MessageInit      `json:"init,omitempty"`
	Segment *MessageSegment   `json:"segment,omitempty"`
	Debug   *MessageDebug     `json:"debug,omitempty"`
	Beat    *MessageHeartBeat `json:"beat,omitempty"`
}

type MessageInit struct {
	Id string `json:"id"` // ID of the init segment
}

type MessageSegment struct {
	Init      string `json:"init"`      // ID of the init segment to use for this segment
	Timestamp int    `json:"timestamp"` // PTS of the first frame in milliseconds
}

type MessageHeartBeat struct {
	Timestamp int `json:"timestamp"`
}

type MessageDebug struct {
	MaxBitrate int `json:"max_bitrate"` // Artificially limit the QUIC max bitrate
}
